chore(wls): Add Windows support

BaptisteFoy opened the pull request:
### What does this PR do?
- Supports APM Workload Selection on Windows by adding `dd-compile-policy.exe` to the build
- Fixes an issue in the Windows `InstallPath` variable, not matching the actual file hierarchy

### Motivation
Making workload Selection work on Windows

### Describe how you validated your changes
Manual QA on a VM

### Additional Notes

BaptisteFoy committed:
chore(wls): Add Windows support

BaptisteFoy committed:
fix windows path computation

bmermet commented on comp/workloadselection/impl/workloadselection_unix.go:
```diff
@@ -0,0 +1,25 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build !windows
```
nit: I don't think that's needed, the file naming should be sufficient.

bmermet commented on comp/workloadselection/impl/workloadselection_windows.go:
```diff
@@ -0,0 +1,26 @@
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build windows
```
idem

BaptisteFoy committed:
address review

BaptisteFoy committed:
fix permissions

BaptisteFoy committed:
Merge branch 'main' into baptiste.foy/WLS/windows-2

bmermet commented on pkg/config/setup/config_windows.go:
```diff
@@ -63,8 +63,8 @@ func osinit() {
 	if _here, err := executable.Folder(); err == nil {
-		InstallPath = filepath.Join(_here, "..", "..")
-		agentFilePath := filepath.Join(InstallPath, "embedded", "agent.exe")
+		InstallPath = filepath.Join(_here, "..")
```
Is this related to policies or an unrelated fix?

bmermet reviewed (approved):
LGTM.
Would it be difficult to write an E2E test for it? Like have the agent write a policy file and check with another user that we can read it.

BaptisteFoy committed:
Merge branch 'main' into baptiste.foy/WLS/windows-2

BaptisteFoy committed:
fix permissions as suggested

BaptisteFoy committed:
fix bazel build according to reviews

BaptisteFoy committed:
address review

BaptisteFoy committed:
address another review

BaptisteFoy committed:
fix(installer): Add proper fips mode (#43805)

This PR adds the `installer` CLI to the supported & tested FIPS binaries

Using the installer in FIPS environments

- E2E test to make sure nothing breaks with the build change
- Automated test for FIPS compliance

Co-authored-by: baptiste.foy <baptiste.foy@datadoghq.com>

rdesgroppes reviewed (approved):
lg!

BaptisteFoy commented:
/merge
